### ISSUE TITLE
Uncaught Exception TypeError: Unsupported operand types: string + int in /var/www/html/glpi9/inc/ipaddress.class.php

### ISSUE THREAD

**Issue by estevamota** [glpi-project/glpi#13992]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Version

9.5.12

### Bug description

Home -> Setup -> Dropdowns -> IP Networks

Name: Test
IP Network: 10.25.0.0/23
Gateway: 10.25.0.2

Returned to white screen

### Relevant log output

```shell
[2023-02-02 13:52:26] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: Unsupported operand types: string + int in /var/www/html/glpi9/inc/ipaddress.class.php at line 435
  Backtrace :
  inc/ipnetwork.class.php:286                        IPAddress->setAddressFromArray()
  inc/ipnetwork.class.php:345                        IPNetwork->prepareInput()
  inc/commondbtm.class.php:1136                      IPNetwork->prepareInputForAdd()
  front/dropdown.common.form.php:59                  CommonDBTM->add()
  front/ipnetwork.form.php:50                        include()
```

### Page URL

http://admrede01.hemoce.ce.gov.br/glpi9/front/ipnetwork.form.php

### Your GLPI setup information

[code]   GLPI 9.5.12 (/glpi9 => /var/www/html/glpi9) Installation mode: TARBALL Current language:en_US

Operating system: Linux 547edfd324fb 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 PHP 8.0.27 apache2handler (Core, FFI, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bz2, calendar, 	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli, 	mysqlnd, openssl, pcre, pdo_mysql, posix, readline, session, shmop, sockets, sodium, standard, sysvmsg, sysvsem, sysvshm, 	tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib) Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files" 	upload_max_filesize="2M"  Software: Apache/2.4.52 (Ubuntu) (Apache/2.4.52 (Ubuntu) Server at admrede01.hemoce.ce.gov.br Port 80) 	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 Server Software: Ubuntu 22.04 	Server Version: 10.6.11-MariaDB-0ubuntu0.22.04.1 	Server SQL Mode:  	Parameters: root@localhost/glpi9 	Host info: Localhost via UNIX socket 	 PHP version is at least 7.2.0 - Perfect! Sessions support is available - Perfect! Allocated memory > 64 Mio - Perfect! mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed intl extension is installed ldap extension is installed apcu extension is installed Zend OPcache extension is installed xmlrpc extension is installed exif extension is installed zip extension is installed bz2 extension is installed sodium extension is installed Database version seems correct (10.6.11) - Perfect! Timezones seems loaded in database The log file has been created successfully. Write access to /var/www/html/glpi9/files/_cache has been validated. Write access to /var/www/html/glpi9/config has been validated. Write access to /var/www/html/glpi9/files/_cron has been validated. Write access to /var/www/html/glpi9/files has been validated. Write access to /var/www/html/glpi9/files/_dumps has been validated. Write access to /var/www/html/glpi9/files/_graphs has been validated. Write access to /var/www/html/glpi9/files/_lock has been validated. Write access to /var/www/html/glpi9/files/_pictures has been validated. Write access to /var/www/html/glpi9/files/_plugins has been validated. Write access to /var/www/html/glpi9/files/_rss has been validated. Write access to /var/www/html/glpi9/files/_sessions has been validated. Write access to /var/www/html/glpi9/files/_tmp has been validated. Write access to /var/www/html/glpi9/files/_uploads has been validated. Write access to /var/www/html/glpi9/marketplace has been validated. Web access to the files directory should not be allowed Check the .htaccess file and the web server configuration.

GLPI_ROOT: /var/www/html/glpi9 GLPI_CONFIG_DIR: /var/www/html/glpi9/config GLPI_VAR_DIR: /var/www/html/glpi9/files GLPI_MARKETPLACE_DIR: /var/www/html/glpi9/marketplace GLPI_USE_CSRF_CHECK: 1 GLPI_CSRF_EXPIRES: 7200 GLPI_CSRF_MAX_TOKENS: 100 GLPI_USE_IDOR_CHECK: 1 GLPI_IDOR_EXPIRES: 7200 GLPI_ALLOW_IFRAME_IN_RICH_TEXT:  GLPI_SERVERSIDE_URL_ALLOWLIST: Array GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org GLPI_INSTALL_MODE: TARBALL GLPI_NETWORK_MAIL: glpi@teclib.com GLPI_NETWORK_SERVICES: https://services.glpi-network.com GLPI_MARKETPLACE_PRERELEASES:  GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1 GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1 GLPI_USER_AGENT_EXTRA_COMMENTS:  GLPI_AJAX_DASHBOARD: 1 GLPI_CALDAV_IMPORT_STATE: 0 GLPI_DEMO_MODE: 0 GLPI_FORCE_EMPTY_SQL_MODE: 1 GLPI_DOC_DIR: /var/www/html/glpi9/files GLPI_CACHE_DIR: /var/www/html/glpi9/files/_cache GLPI_CRON_DIR: /var/www/html/glpi9/files/_cron GLPI_DUMP_DIR: /var/www/html/glpi9/files/_dumps GLPI_GRAPH_DIR: /var/www/html/glpi9/files/_graphs GLPI_LOCAL_I18N_DIR: /var/www/html/glpi9/files/_locales GLPI_LOCK_DIR: /var/www/html/glpi9/files/_lock GLPI_LOG_DIR: /var/www/html/glpi9/files/_log GLPI_PICTURE_DIR: /var/www/html/glpi9/files/_pictures GLPI_PLUGIN_DOC_DIR: /var/www/html/glpi9/files/_plugins GLPI_RSS_DIR: /var/www/html/glpi9/files/_rss GLPI_SESSION_DIR: /var/www/html/glpi9/files/_sessions GLPI_TMP_DIR: /var/www/html/glpi9/files/_tmp GLPI_UPLOAD_DIR: /var/www/html/glpi9/files/_uploads GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/ GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/ GLPI_I18N_DIR: /var/www/html/glpi9/locales GLPI_VERSION: 9.5.12 GLPI_SCHEMA_VERSION: 9.5.7 GLPI_MIN_PHP: 7.2.0 GLPI_YEAR: 2022

htmlawed/htmlawed version 1.2.5 in (/var/www/html/glpi9/vendor/htmlawed/htmlawed) phpmailer/phpmailer version 6.6.5 in (/var/www/html/glpi9/vendor/phpmailer/phpmailer/src) simplepie/simplepie version 1.5.6 in (/var/www/html/glpi9/vendor/simplepie/simplepie/library) tecnickcom/tcpdf version 6.3.5 in (/var/www/html/glpi9/vendor/tecnickcom/tcpdf) michelf/php-markdown in (/var/www/html/glpi9/vendor/michelf/php-markdown/Michelf) true/punycode in (/var/www/html/glpi9/vendor/true/punycode/src) iamcal/lib_autolink in (/var/www/html/glpi9/vendor/iamcal/lib_autolink) sabre/dav in (/var/www/html/glpi9/vendor/sabre/dav/lib/DAV) sabre/http in (/var/www/html/glpi9/vendor/sabre/http/lib) sabre/uri in (/var/www/html/glpi9/vendor/sabre/uri/lib) sabre/vobject in (/var/www/html/glpi9/vendor/sabre/vobject/lib) laminas/laminas-cache in (/var/www/html/glpi9/vendor/laminas/laminas-cache/src) laminas/laminas-i18n in (/var/www/html/glpi9/vendor/laminas/laminas-i18n/src) laminas/laminas-serializer in (/var/www/html/glpi9/vendor/laminas/laminas-serializer/src) monolog/monolog in (/var/www/html/glpi9/vendor/monolog/monolog/src/Monolog) sebastian/diff in (/var/www/html/glpi9/vendor/sebastian/diff/src) elvanto/litemoji in (/var/www/html/glpi9/vendor/elvanto/litemoji/src) symfony/console in (/var/www/html/glpi9/vendor/symfony/console) scssphp/scssphp in (/var/www/html/glpi9/vendor/scssphp/scssphp/src) laminas/laminas-mail in (/var/www/html/glpi9/vendor/laminas/laminas-mail/src/Protocol) laminas/laminas-mime in (/var/www/html/glpi9/vendor/laminas/laminas-mime/src) rlanvin/php-rrule in (/var/www/html/glpi9/vendor/rlanvin/php-rrule/src) blueimp/jquery-file-upload in (/var/www/html/glpi9/vendor/blueimp/jquery-file-upload/server/php) ramsey/uuid in (/var/www/html/glpi9/vendor/ramsey/uuid/src) psr/log in (/var/www/html/glpi9/vendor/psr/log/Psr/Log) psr/simple-cache in (/var/www/html/glpi9/vendor/psr/simple-cache/src) mexitek/phpcolors in (/var/www/html/glpi9/vendor/mexitek/phpcolors/src/Mexitek/PHPColors) guzzlehttp/guzzle in (/var/www/html/glpi9/vendor/guzzlehttp/guzzle/src) guzzlehttp/psr7 in (/var/www/html/glpi9/vendor/guzzlehttp/psr7/src) wapmorgan/unified-archive in (/var/www/html/glpi9/vendor/wapmorgan/unified-archive/src) paragonie/sodium_compat in (/var/www/html/glpi9/vendor/paragonie/sodium_compat/src)

Not active

Way of sending emails: PHP

 

 
[/code]

nformation about system installation & configuration
[Check if a New Version is Available](http://admrede01.hemoce.ce.gov.br/glpi9/front/config.form.php?check_version)
[code]
 
GLPI 9.5.12 (/glpi9 => /var/www/html/glpi9)
Installation mode: TARBALL
Current language:en_US

Server
 
Operating system: Linux 547edfd324fb 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64
PHP 8.0.27 apache2handler (Core, FFI, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bz2, calendar,
	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli,
	mysqlnd, openssl, pcre, pdo_mysql, posix, readline, session, shmop, sockets, sodium, standard, sysvmsg, sysvsem, sysvshm,
	tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="30" memory_limit="128M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="2M" 
Software: Apache/2.4.52 (Ubuntu) (Apache/2.4.52 (Ubuntu) Server at admrede01.hemoce.ce.gov.br Port 80)
	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36
Server Software: Ubuntu 22.04
	Server Version: 10.6.11-MariaDB-0ubuntu0.22.04.1
	Server SQL Mode: 
	Parameters: root@localhost/glpi9
	Host info: Localhost via UNIX socket
	
PHP version is at least 7.2.0 - Perfect!PHP version is at least 7.2.0 - Perfect!
Sessions support is available - Perfect!Sessions support is available - Perfect!
Allocated memory > 64 Mio - Perfect!Allocated memory > 64 Mio - Perfect!
mysqli extension is installedmysqli extension is installed
ctype extension is installedctype extension is installed
fileinfo extension is installedfileinfo extension is installed
json extension is installedjson extension is installed
mbstring extension is installedmbstring extension is installed
iconv extension is installediconv extension is installed
zlib extension is installedzlib extension is installed
curl extension is installedcurl extension is installed
gd extension is installedgd extension is installed
simplexml extension is installedsimplexml extension is installed
intl extension is installedintl extension is installed
ldap extension is installedldap extension is installed
apcu extension is installedapcu extension is installed
Zend OPcache extension is installedZend OPcache extension is installed
xmlrpc extension is installedxmlrpc extension is installed
exif extension is installedexif extension is installed
zip extension is installedzip extension is installed
bz2 extension is installedbz2 extension is installed
sodium extension is installedsodium extension is installed
Database version seems correct (10.6.11) - Perfect!Database version seems correct (10.6.11) - Perfect!
Timezones seems loaded in databaseTimezones seems loaded in database
The log file has been created successfully.The log file has been created successfully.
Write access to /var/www/html/glpi9/files/_cache has been validated.Write access to /var/www/html/glpi9/files/_cache has been validated.
Write access to /var/www/html/glpi9/config has been validated.Write access to /var/www/html/glpi9/config has been validated.
Write access to /var/www/html/glpi9/files/_cron has been validated.Write access to /var/www/html/glpi9/files/_cron has been validated.
Write access to /var/www/html/glpi9/files has been validated.Write access to /var/www/html/glpi9/files has been validated.
Write access to /var/www/html/glpi9/files/_dumps has been validated.Write access to /var/www/html/glpi9/files/_dumps has been validated.
Write access to /var/www/html/glpi9/files/_graphs has been validated.Write access to /var/www/html/glpi9/files/_graphs has been validated.
Write access to /var/www/html/glpi9/files/_lock has been validated.Write access to /var/www/html/glpi9/files/_lock has been validated.
Write access to /var/www/html/glpi9/files/_pictures has been validated.Write access to /var/www/html/glpi9/files/_pictures has been validated.
Write access to /var/www/html/glpi9/files/_plugins has been validated.Write access to /var/www/html/glpi9/files/_plugins has been validated.
Write access to /var/www/html/glpi9/files/_rss has been validated.Write access to /var/www/html/glpi9/files/_rss has been validated.
Write access to /var/www/html/glpi9/files/_sessions has been validated.Write access to /var/www/html/glpi9/files/_sessions has been validated.
Write access to /var/www/html/glpi9/files/_tmp has been validated.Write access to /var/www/html/glpi9/files/_tmp has been validated.
Write access to /var/www/html/glpi9/files/_uploads has been validated.Write access to /var/www/html/glpi9/files/_uploads has been validated.
Write access to /var/www/html/glpi9/marketplace has been validated.Write access to /var/www/html/glpi9/marketplace has been validated.
Web access to the files directory should not be allowed Check the .htaccess file and the web server configuration.Web access to the files directory should not be allowed
Check the .htaccess file and the web server configuration.

GLPI constants
 
GLPI_ROOT: /var/www/html/glpi9
GLPI_CONFIG_DIR: /var/www/html/glpi9/config
GLPI_VAR_DIR: /var/www/html/glpi9/files
GLPI_MARKETPLACE_DIR: /var/www/html/glpi9/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_SERVERSIDE_URL_ALLOWLIST: Array
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/www/html/glpi9/files
GLPI_CACHE_DIR: /var/www/html/glpi9/files/_cache
GLPI_CRON_DIR: /var/www/html/glpi9/files/_cron
GLPI_DUMP_DIR: /var/www/html/glpi9/files/_dumps
GLPI_GRAPH_DIR: /var/www/html/glpi9/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/www/html/glpi9/files/_locales
GLPI_LOCK_DIR: /var/www/html/glpi9/files/_lock
GLPI_LOG_DIR: /var/www/html/glpi9/files/_log
GLPI_PICTURE_DIR: /var/www/html/glpi9/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/www/html/glpi9/files/_plugins
GLPI_RSS_DIR: /var/www/html/glpi9/files/_rss
GLPI_SESSION_DIR: /var/www/html/glpi9/files/_sessions
GLPI_TMP_DIR: /var/www/html/glpi9/files/_tmp
GLPI_UPLOAD_DIR: /var/www/html/glpi9/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /var/www/html/glpi9/locales
GLPI_VERSION: 9.5.12
GLPI_SCHEMA_VERSION: 9.5.7
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2022

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/var/www/html/glpi9/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.6.5 in (/var/www/html/glpi9/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/var/www/html/glpi9/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/var/www/html/glpi9/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/html/glpi9/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/glpi9/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/glpi9/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/glpi9/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/glpi9/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/glpi9/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/glpi9/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/var/www/html/glpi9/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/var/www/html/glpi9/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/var/www/html/glpi9/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/var/www/html/glpi9/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/glpi9/vendor/sebastian/diff/src)
elvanto/litemoji in (/var/www/html/glpi9/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/glpi9/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/glpi9/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/glpi9/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/glpi9/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/glpi9/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/html/glpi9/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/html/glpi9/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/glpi9/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/glpi9/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/var/www/html/glpi9/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/glpi9/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/html/glpi9/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/var/www/html/glpi9/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/glpi9/vendor/paragonie/sodium_compat/src)

SQL replicas
 
Not active

Notifications
 
Way of sending emails: PHP

Mails receivers
 

Plugins list
 

[/code]
To copy/paste in your support request

**Comment by trasher** [comment]
GLPI 9.5 is no longer supported, upgrade to latest stable release.

**Comment by cedric-anne** [comment]
Problem have probably been fixed by commit 95f2855fdeb74d7fc3b4c373b25900f5f034041f in 10.0 version.
